plugins/blog-plugin.js

```diff
@@ -0,0 +1,44 @@
+const fs = require('node:fs');
+
+const blogPluginExports = require('@docusaurus/plugin-content-blog');
+
+const defaultBlogPlugin = blogPluginExports.default;
+
+async function blogPluginEnhanced(...pluginArgs) {
+  const blogPluginInstance = await defaultBlogPlugin(...pluginArgs);
+
+  const dir = '.docusaurus';
+
+  return {
+    ...blogPluginInstance,
+    contentLoaded: async function (data) {
+      // here we decide the number of posts to write to recent-posts.json
+      let recentPosts = [...data.content.blogPosts]
+        // Only show published posts.
+        .filter((p) => !p.metadata.unlisted)
+        .slice(0, 5);
+
+      recentPosts = recentPosts.map((p) => {
+        return {
+          id: p.id,
+          metadata: p.metadata,
+        };
+      });
+
+      fs.mkdirSync(dir, {
+        recursive: true, // Avoid error if directory already exists.
+      });
+      // this is to update the recent-posts.json
+      // the same json that is imported in the index.jsx (Home page)
+      const fd = fs.openSync(`${dir}/recent-posts.json`, 'w');
+      fs.writeSync(fd, JSON.stringify(recentPosts));
```

Comment:
I don't like this writing directly to the file system. Did you have a look at [this example](https://github.com/facebook/docusaurus/issues/4138#issuecomment-1703083513)? 

I also [asked](https://github.com/facebook/docusaurus/discussions/6423#discussioncomment-10235638) if there's a better way to do that in the most recent version of Docusaurus.

Reply:
Understood, will look into this.